src/test/java/bio/terra/endpoints/CLICommandTests.java

```diff
@@ -0,0 +1,284 @@
+     * @throws IOException
+     */
+    public Map<String, Object> retrieveDatasetHttpGET(String token, String datasetId) throws IOException {
+        System.out.println("***********************************************");
+        System.out.println("HTTP request api/repository/v1/datasets/{id}");
+
+        // endpoint information
+        String endpointName = "api/repository/v1/datasets/" + datasetId;
+        String endpointType = "GET";
+
+        // make request using Java HTTP library
+        Map<String, Object> javaHttpResponse = CLITestingUtils.sendJavaHttpRequest(
+                CLITestingConfig.dataRepoURL + endpointName, endpointType, token, null);
+
+        // log the response to stdout
+        System.out.println("javaHttpResponse: " + javaHttpResponse + "\n");
+
+        // check that the status code is success
+        Assert.assertEquals(200, javaHttpResponse.get("statusCode"));
+
+        // return the dataset details as a map
+        return javaHttpResponse;
+    }
+
+    /**
+     * CLI command : jc dataset show
+     * Repository API : GET : enumerateDatasets
+     * @throws IOException
+     */
+    public void datasetShowTest(Map<String, Object> datasetDetails, String expectedOutputFilename)
+            throws IOException {
+        String datasetName = datasetDetails.get("name").toString();
+        System.out.println("***********************************************");
+        System.out.println("jc dataset show " + datasetName);
+
+        // call CLI command in a separate process
+        List<String> cmdArgs = new ArrayList<String>(new ArrayList<>(Arrays.asList(
+                new String[]{"dataset", "show", datasetName})));
+        List<String> cliCmdResponse = CLITestingUtils.callCLICommand(cmdArgs);
+
+        // read in the expect CLI command output from a file
+        List<String> cliCmdExpectedResponse = CLITestingUtils.readCLIExpectedOutput(expectedOutputFilename);
+
+        // log the response to stdout
+        System.out.println("cliCmdResponse: " + cliCmdResponse + "\n");
+
+        // need to replace %xyz% variables in the CLI expected response with values from the Java HTTP response
+        String id = datasetDetails.get("id").toString();
+        String createdDate = datasetDetails.get("createdDate").toString();
+        ArrayList<Object> assets =
+                (ArrayList<Object>)((Map<String, Object>)datasetDetails.get("schema")).get("assets");
+        ArrayList<Object> assetTables = (ArrayList<Object>)((Map<String, Object>)assets.get(0)).get("tables");
+        ArrayList<String> assetTableNames = new ArrayList<>();
+        for (int ctr = 0; ctr < assetTables.size(); ctr++) {
+            String tableName = ((Map<String, Object>)assetTables.get(ctr)).get("name").toString();
+            assetTableNames.add(tableName);
+        }
+
+        // check that the responses match
+        for (int ctr = 0; ctr < cliCmdResponse.size(); ctr++) {
```

Comment:
Do you think the compare would be easier if the output was JSON instead of text?

Reply:
Yes, I think it would be easier to compare. But the CLI output is intended to be human-readable, not necessarily machine-parseable. So, if it's just to make testing easier I don't think that's a good reason to change the output format.